content/blog/2016/2016-06-30-html-publisher-plugin.adoc

```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+title: "Publishing HTML Reports in Pipeline"
+tags:
+- pipeline
+- plugins
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at Cloudbees.
+
+Most projects need more that just JUnit result reporting.  Rather than writing a
+custom plugin for each type of report, we can use the
+link:https://wiki.jenkins-ci.org/display/JENKINS/HTML+Publisher+Plugin[HTML Publisher Plugin].
+
+=== Let's Make This Quick
+
+I've found a ruby project,
+link:https://github.com/reiseburo/hermann[hermann], I'd like to build using Jenkins Pipeline. I'd
+also like to have the code coverage results published with each build job.  I could
+write a plugin to publish this data, but I'm in a bit of hurry and it the build
+already creates an html report file.
+
+=== Simple Build
+I'm going to use the
+link:https://wiki.jenkins-ci.org/display/JENKINS/HTML+Publisher+Plugin[HTML Publisher Plugin]
+to add the HTML-formatted code coverage report to my builds.  Here's a simple
+pipeline for building the link:https://github.com/reiseburo/hermann[hermann]
+project.
+
+[source,groovy]
+----
+stage 'Build'
+
+node {
+  // Checkout
+  checkout scm
+
+  // Use a particular version of ruby (configured by rvm) to run build
+  sh ('''
+  # install required bundles
+  bundle install
+
+  # compile native components
+  bundle exec rake compile
+
+  # build the package
+  bundle exec rake build
+
+  # run tests and coverage
+  bundle exec rake spec
+  ''')
+
+  // Archive the built artifacts
+  archive (includes: 'pkg/*.gem')
+}
+----
+
+Simple enough, it builds, runs tests, and archives the package.
+image::/images/post-images/2016-06-30/run-1.png[Job Run Without Report Link, role=center]
```

Comment:
I think this needs a newline above it for the `image` macro to be rendered properly (GitHub doesn't think there's an image here)